stable/spark-history-server/templates/configmap.yaml

```diff
@@ -16,19 +16,19 @@ data:
   {{ $key }}: {{ $val | quote }}
   {{- end }}
   {{- if .Values.pvc.enablePVC }}
-  {{- range $key, $val := .Values.pvc }}
+  {{- range $key, $val := omit .Values.pvc "enablePVC" }}
```

Comment:
That's for the PR. I'm not familiar with the `omit` syntax. Could you explain what this line does:
`omit .Values.pvc "enablePVC"`? Iterating through `.Values.pvc` only if `enablePVC` is true?

Reply:
Sure, [`omit` is from the Sprig library](https://masterminds.github.io/sprig/dicts.html#omit), it returns a dict without the specified keys (in this case `enablePVC`)

Reply:
Okay, so that means the user needs the Sprig dependency to use this chart going forward? And I'm not sure about the problem you are trying to solve. In the PR description, you said you want to remove "duplicate configmap keys". Is `enableGCS` duplicated?

Reply:
[Sprig comes default with Helm](https://helm.sh/docs/howto/charts_tips_and_tricks/#know-your-template-functions), so it shouldn't require anything from users of the Chart.

And yes, [there are four hardcoded keys at the top](https://github.com/helm/charts/blob/fb1752fd9294cdba0c499a7b7736daea49bb8c35/stable/spark-history-server/templates/configmap.yaml#L11-L14), and then the four loops below will output those keys again, that's why I'm omitting them from each loop.

Duplicate keys in a map aren't technically valid Yaml, even though Helm is able to handle them. See this section in the [Yaml spec](https://yaml.org/spec/1.2/spec.html#id2759572), specifically:

```
JSON's RFC4627 requires that mappings keys merely “SHOULD” be unique, while YAML insists they “MUST” be. Technically, YAML therefore complies with the JSON spec, choosing to treat duplicates as an error. In practice, since JSON is silent on the semantics of such duplicates, the only portable JSON files are those with unique keys, which are therefore valid YAML files. 
```

Here's the output of the config map's data from current master (comments mine):
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: spark-history-server
  labels:
    app.kubernetes.io/name: spark-history-server
    helm.sh/chart: spark-history-server-1.1.1
    app.kubernetes.io/instance: spark-history-server
    app.kubernetes.io/managed-by: Helm
data:
  # Four hardcoded keys
  enablePVC: "true"
  enableGCS: "false"
  enableS3: "false"
  enableWASBS: "false"
  # Generated by loops
  enablePVC: "true"
  eventsDir: "/"
  existingClaimName: "spark-history"
```

And here's the output with this PR applied:

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: ddm-spark-history-server
  labels:
    app.kubernetes.io/name: spark-history-server
    helm.sh/chart: spark-history-server-1.1.1
    app.kubernetes.io/instance: ddm-spark-history-server
    app.kubernetes.io/managed-by: Helm
data:
  # Four hardcoded keys
  enablePVC: "true"
  enableGCS: "false"
  enableS3: "false"
  enableWASBS: "false"
  # Generated by loops
  eventsDir: "/"
  existingClaimName: "spark-history"
```

Sorry, should have put more context in the original PR, I'll move some of this there too.

Reply:
Thank you for the detailed clarification. Makes sense to me